utils/fixed_decimal/src/decimal.rs

```diff
@@ -552,6 +552,197 @@ impl FixedDecimal {
         self.check_invariants();
     }
 
+    /// Increments the digits by 1. if the digits are empty, it will add
+    /// an element with value 1 and assign the value of the upper magnitude to
+    /// the magnitude.
+    fn increment_abs_by_one(&mut self) {
+        for i in (0..self.digits.len()).rev() {
+            self.digits[i] += 1;
+            if self.digits[i] < 10 {
+                return;
+            }
+            self.digits[i] = 0;
+        }
+
+        // Still there is a carry, add `1` in the beginning of the digits.
```

Comment:
Suggestion: If this happens, the array is always filled with zeros. You should just replace the whole array with a single entry of `1` and update the magnitude fields as needed.

Reply:
now, we just push a `1`